src/lib/support/Pool.h

```diff
@@ -166,6 +186,46 @@ struct HeapObjectList : HeapObjectListNode
 
 } // namespace internal
 
+/// Provides iteration over active objects in a Bitmap pool.
+///
+/// Creating and releasing items within a pool does not invalidate
+/// an iterator, however there are no guarantees which objects the
+/// iterator will return (i.e. newly created objects while iterating
+/// may be visible or not to the iterator depending where they are
+/// allocated).
+///
+/// You are not prevented from releasing the object the iterator
+/// currently points at. In that case, iterator should be advanced.
+template <class T>
+class BitmapActiveObjectIterator
+{
+public:
+    using value_type = T;
+    using pointer    = T *;
+    using reference  = T &;
+
+    explicit BitmapActiveObjectIterator(internal::StaticAllocatorBitmap * pool, size_t idx) : mPool(pool), mIndex(idx) {}
+    BitmapActiveObjectIterator() {}
```

Comment:
Do we even need this constructor?

Reply:
I have a unit test that says we have a well defined "default iterator" (points at the end).
The intent for this was to ensure that one can just create an iterator without initializing it.